www/app/Chat.tsx

```diff
@@ -305,13 +299,46 @@ export default function Chat({
 
     try {
       // Get thought stream
-      const thoughtStream = await fetchStream(
+      const thoughtResponse = await fetchStream(
         'thought',
         messageToSend,
         conversationId!
       );
-      if (!thoughtStream) throw new Error('Failed to get thought stream');
 
+      if (thoughtResponse.status === 402) {
+        setCanSend(false);
+        mutateMessages(
+          messages => [
+            ...(messages?.slice(0, -1) || []),
+            {
+              content: '',
+              isUser: false,
+              id: '',
+              metadata: {},
+              error: 402
+            }
+          ],
+          { revalidate: false }
+        );
+        Swal.fire({
+          title: 'Subscription Required',
+          text: 'You have no active subscription. Subscribe to continue using Bloom!',
+          icon: 'warning',
+          confirmButtonColor: '#3085d6',
+          confirmButtonText: 'Subscribe',
+          showCancelButton: false,
+        }).then((result) => {
+          if (result.isConfirmed) {
+            router.push('/settings');
+          }
+        });
+        return;
```

Comment:
Don't need extra logic on front-end as the error should only occur on instances of users trying to bypass the UI